.github/workflows/test.yml

```diff
@@ -0,0 +1,92 @@
+name: test
+on: pull_request
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+    env:
+      DATABASE_URL: postgresql://hatchet:hatchet@127.0.0.1:5431/hatchet
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v2
+        with:
+          version: '23.x'
+
+      - name: Install Task
+        uses: arduino/setup-task@v1
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+
+      - name: Compose
+        run: docker compose up -d
```

Comment:
the `generate` needs compose because sqlc needs an sql file which is generated by prisma migrate which needs a live db